crypto/evp_extra/scrypt.c

```diff
@@ -127,6 +134,11 @@ static void scryptROMix(block_t *B, uint64_t r, uint64_t N, block_t *T,
     }
     scryptBlockMix(B, T, r);
   }
+#ifdef OPENSSL_BIG_ENDIAN
+  for(size_t i = 0; i < (2 * r * SCRYPT_BLOCK_WORD_CNT); i++) {
+    CRYPTO_store_u32_le(&B->words[i], B->words[i]);
+  }
+#endif
```

Comment:
Is it possible to update scryptBlockMix or some other function to write out B in the correct order so you don't have to fix it for every call to scryptROMix?

Reply:
`scryptROMix` is only called [from a for-loop](https://github.com/aws/aws-lc/blob/main/crypto/evp_extra/scrypt.c#L196) in the top-level function. A different set of blocks is passed each time, so I don't see much opportunity to improve performance by moving it up.
```
  for (uint64_t i = 0; i < p; i++) {
    scryptROMix(B + 2 * r * i, r, N, T, V);
  }
```

I also don't think performance would be improved by moving it down, since each word seems to be overwritten numerous times per invocation.

I checked [OpenSSL's implementation](https://github.com/openssl/openssl/blob/143ca66cf00c88950d689a8aa0c89888052669f4/providers/implementations/kdfs/scrypt.c#L392), and they appear to perform the big-endian tranformation at roughly the same location in their logic.